pkg/apis/integreatly/v1alpha1/types/types.go

```diff
@@ -18,12 +20,23 @@ type SecretRef struct {
 	Namespace string `json:"namespace,omitempty"`
 }
 
+type Env struct {
+	Name  string `json:"name"`
+	Value string `json:"value"`
+}
+
+type Labels struct {
+	ProductName string `json:"productname"`
+}
+
 // ResourceTypeSpec Represents the basic information required to provision a resource type
 // +k8s:openapi-gen=true
 type ResourceTypeSpec struct {
 	Type      string     `json:"type"`
 	Tier      string     `json:"tier"`
 	SecretRef *SecretRef `json:"secretRef"`
+	Env       *Env       `json:"env"`
+	Labels    *Labels    `json:"labels"`
```

Comment:
Personally I think these fields should be optional. If the labels are there, we can use them to tag our resources. If the organisation tag is there, we can use that too, or otherwise set a reasonable/empty fallback value.

Reply:
I agree here, if we enforce these labels, we will lose backwards compatible, this will force us to have to create a new version of the CRD `v1alpha2` to avoid completely breaking work already carried out in the intly operator.

Reply:
I have no problem adding omit on empty to these.